test/node/native/native.node.test.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * @vitest-environment jsdom
+ */
+import { http } from 'msw'
+import { setupServer } from 'msw/native'
```

Comment:
can you please tell me more about when this test would fail? Since we are in Node, events module will always exist so I think this test isn't testing anything.

Reply:
we can also model the env by ourselves if vitest allows different vitest configs per environment. we could create an `alias` option to `node:events` to resolve it with a module that throws, thus emulating what happens in react native. that can be an approach to consider.

Reply:
This mostly ensures that the server works when we don't pass node events through. 

Since they're more of a dependency injection in this model, since we're using msw/native and not msw/node we don't even attempt to import them

I'd love a better way to test this, but not sure how best to set that up currently

Reply:
the closer to the actual environment we model the test, the better that test will be. in our case, we need a test environment that fails when importing the `node:events` module. then we ensure that when using msw/native, we can still intercept and mock things.